R/uvhydrograph-render.R

```diff
@@ -168,20 +174,22 @@ createPrimaryPlot <- function(primaryData, primaryInfo){
   primaryData <-
     primaryData[c(grep("^uncorr_UV$", names(primaryData)),
                   grep("^uncorr_UV$", names(primaryData), invert = TRUE))]
-
-  # reorder so that corrections are plotted last
-  primaryData <-
-    primaryData[c(grep("^series_corr$", names(primaryData), invert = TRUE),
-                  grep("^series_corr$", names(primaryData)))]
   
+  lims <- primaryInfo[[grep("lims_UV", names(primaryInfo))]]
   # add data to plot
+  # TODO: Once we've fully deconstructed primaryData into individual calls, this loop might become single explicit calls
   for (i in grep("^appr_.+_uv", names(primaryData), invert = TRUE)) {
     plot_object <-
         AddToGsplot(plot_object, getPrimaryPlotConfig(primaryData[i], primaryInfo$plotStartDate, primaryInfo$plotEndDate, 
                 primaryInfo$primary_lbl, primaryInfo$reference_lbl, paste("Comparison", primaryInfo$comp_UV_TS_lbl, "@", primaryInfo$comp_UV_lbl),
-                sides, ylims, primaryInfo[[grep("lims_UV", names(primaryInfo))]]))
+                sides, ylims, lims))
   }
   
+  # corrections have been pulled out of primaryData and are their own top level object. Need to get it's own style info
+  # and add to plot. NOTE: this is out of original order for now.
+  plot_object <- AddToGsplot(plot_object, getCorrectionsPlotConfig(corrections, primaryInfo$plotStartDate, primaryInfo$plotEndDate, 
```

Comment:
I like this function for the `do.call` stuff. But agree that explicit calls (such as what CORR does) would be better.